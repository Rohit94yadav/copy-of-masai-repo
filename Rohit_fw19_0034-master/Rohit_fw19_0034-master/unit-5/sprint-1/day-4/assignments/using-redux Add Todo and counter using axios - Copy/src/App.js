
import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import './App.css';
import { handleAddCount, handleReduceCount } from './Redux/Action';
import { Store } from './Redux/Store';

function App() {

//   const {dispatch, subscribe} = Store;

// const {count} = Store.getState();

// const [update, setUpdate] = useState(0)

// subscribe(()=>{
//   setUpdate(prev => prev+1)
// })

const count = useSelector((state)=> state.count)
const dispatch = useDispatch();

const handleAdd = ()=>{
  dispatch(handleAddCount(1))
}

const handleReduce = ()=>{
  dispatch(handleReduceCount(2))
}

  return (
    <div className="App">
     <h1>Counter {count}</h1>
     <button onClick={handleAdd}>ADD</button>
     <button onClick={handleReduce}>REDUCE</button>
    </div>
  );
}

export default App;
