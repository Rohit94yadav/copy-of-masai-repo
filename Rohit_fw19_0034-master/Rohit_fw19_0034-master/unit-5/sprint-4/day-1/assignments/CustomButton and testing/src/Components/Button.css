.btn{
    color: white;
    cursor: pointer;
    font-size: 30px;

}

.red{
    background-color: red;
}

.ghost{
    border: none;
}

.green{
    background-color: green;
}

.bordered{
    border: 1px solid yellow;
}