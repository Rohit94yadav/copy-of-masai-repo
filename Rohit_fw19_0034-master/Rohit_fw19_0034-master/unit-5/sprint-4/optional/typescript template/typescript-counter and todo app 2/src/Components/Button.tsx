import React from 'react'
interface ButtonProps{
    buttonText: string,
    handleClick: ()=> void
}
const Button = (props : ButtonProps) => {
    const {buttonText, handleClick} = props;

  return (
    <div>

    <button onClick={handleClick}>{buttonText}</button>
    </div>
  )
}

export default Button
