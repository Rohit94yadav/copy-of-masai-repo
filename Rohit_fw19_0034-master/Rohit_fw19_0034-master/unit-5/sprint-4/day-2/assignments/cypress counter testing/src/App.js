

import axios from 'axios';
import { useEffect, useState } from 'react';
import './App.css';




function App() {

  const [counter, setCounter] = useState(0)

  const incCount = ()=>{
    axios.post("http://localhost:8080/counter", {value : counter+1})
    .then((r)=> setCounter(r.data.value))
  }

  const decCount = ()=>{
    setCounter((prev) => prev-1)
  }

  const fetchData = ()=>{
    axios.get("http://localhost:8080/counter")
    .then(({data}) => {
      setCounter(data.value)
    })
    .catch((e)=>{
      console.log(e)
    })
  }

  useEffect(()=>{
    fetchData()
  },[])

  return (
    <div className="App">
      <h1>Counter App</h1>
      <button className= "addButton" onClick={incCount} >Add</button>
        <h2>The current count is {counter}</h2>
        <button className = "reduceButton" onClick={decCount} >Reduce</button>
    </div>
  );
}

export default App;
