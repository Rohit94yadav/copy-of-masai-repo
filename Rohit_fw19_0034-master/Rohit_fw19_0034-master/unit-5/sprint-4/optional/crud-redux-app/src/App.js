

import './App.css';
import MainRoutes from './Pages/MainRoutes';

import { Store } from './Redux/Store';

function App() {





  return (
    <div className="App">
     <MainRoutes/>
     
    </div>
  );
}

export default App;
