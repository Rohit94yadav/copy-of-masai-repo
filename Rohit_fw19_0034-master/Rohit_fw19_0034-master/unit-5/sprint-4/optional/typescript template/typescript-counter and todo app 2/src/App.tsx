import React from 'react';

import './App.css';
import Counter from './Components/Counter';
import Todo from './Components/Todo';

function App() {
  return (
   <div>
      <Counter/>
      <Todo/>
   </div> 
   
  );
}

export default App;
