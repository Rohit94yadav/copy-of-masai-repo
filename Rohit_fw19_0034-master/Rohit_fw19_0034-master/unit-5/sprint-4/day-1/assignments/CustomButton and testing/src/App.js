
import { useState } from 'react';
import './App.css';
import Button from './Components/Button';


function App() {

  const [theme, setTheme] = useState("light")

  return (
    <div className="App">
      <h3>Current theme is {theme}</h3>
      <Button colorScheme={"green"} 
      onClick = {()=>{setTheme(theme === "light" ? "dark" :  "light")}}
      >Click Me</Button>
    </div>
  );
}

export default App;
