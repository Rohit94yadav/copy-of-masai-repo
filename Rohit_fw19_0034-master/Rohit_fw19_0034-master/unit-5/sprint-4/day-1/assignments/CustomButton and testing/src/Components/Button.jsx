import React from 'react'
import "./Button.css"
const Button = ({children, colorScheme, variant, onClick}) => {
  return (
    <button data-testid= "customButton"
    className={`btn ${colorScheme || "red"} ${variant || "bordered"}`} 
    onClick={onClick} >
      {children}
    </button>
  )
}

export default Button
