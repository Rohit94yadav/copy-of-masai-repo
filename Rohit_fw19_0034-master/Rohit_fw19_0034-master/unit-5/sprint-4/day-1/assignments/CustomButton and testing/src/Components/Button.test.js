import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import App from "../App";
import Button from "./Button"
import renderer from "react-test-renderer";


describe("Testing the custom button cumponent", ()=>{
    it("should be in the DOM tree", ()=>{
        render(<Button>Testing Button</Button>);
       let button = screen.getByText("Testing Button");

       expect(button).toBeInTheDocument();
    });

    it("should have a button with test Click Me in app.js", ()=>{
        render(<App/>);
        // screen.debug();
        let button = screen.getByTestId("customButton");
        expect(button).toHaveTextContent("Click Me");
    });

    it("should check if the label is present as passed through the props", ()=>{
        render(<Button>Testing</Button>);
        // screen.debug();
        let button = screen.getByTestId("customButton");
        expect(button).toHaveTextContent("Testing");
    });

    it("should be empty DOM if children props is not passed", ()=>{
        render(<Button></Button>);
        // screen.debug();
        let button = screen.getByTestId("customButton");
        expect(button).toBeEmptyDOMElement();
    });

    it("should toggle the theme when the button is clicked", ()=>{
        render(<App/>);
        // screen.debug();
        let button = screen.getByTestId("customButton");
        let h3Tag = screen.getByText("Current theme is light");
        expect(h3Tag).toHaveTextContent("light");
        userEvent.click(button);
        expect(h3Tag).toHaveTextContent("dark")
    });

    it("should invoke the onClick function", ()=>{
        const mockFun = jest.fn();

        render(<Button onClick={mockFun} >Checking onClick</Button>);
        // screen.debug();
        let button = screen.getByTestId("customButton");
        userEvent.click(button);
        userEvent.click(button);
        // expect(mockFun).toBeCalled();
        expect(mockFun).toBeCalledTimes(2)
    });

    it("should match the snapshot", ()=>{

        const tree = renderer.create(<Button colorScheme={"green"} 
        variant={"ghost"}>Custom Button</Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });


});