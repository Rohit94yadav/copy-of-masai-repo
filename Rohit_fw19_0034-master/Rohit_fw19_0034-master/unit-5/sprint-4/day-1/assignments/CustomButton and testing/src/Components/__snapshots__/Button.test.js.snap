// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the custom button cumponent should match the snapshot 1`] = `
<button
  className="btn green ghost"
  data-testid="customButton"
>
  Custom Button
</button>
`;
